test/functional/lua/vim_spec.lua

```diff
@@ -1205,8 +1205,13 @@ describe('lua stdlib', function()
     ]])
     eq(true, exec_lua([[return next(vim.fn.FooFunc(3)) == nil ]]))
     eq(3, eval('g:test'))
-    -- compat: nvim_call_function uses "special" value for empty dict
-    eq(true, exec_lua([[return next(vim.api.nvim_call_function("FooFunc", {5})) == true ]]))
+    eq(
+      { {}, true, true },
+      exec_lua([[
+    local ret = vim.api.nvim_call_function("FooFunc", {5})
+    return {ret, vim.tbl_isempty(ret), next(ret) == nil}
```

Comment:
these are a bit redundant, it is enough to use `return vim.tbl_isempty(vim.api.nvim_call_function("FooFunc", {5}))` (tbl_isempty throws an error for a non-table)

Reply:
Done in [c6902b4](https://github.com/neovim/neovim/pull/29659/commits/c6902b4c979e7a37c8ffb8d485ba667c1a159e4a)